apa102/src/main/java/com/google/android/things/contrib/driver/apa102/Apa102.java

```diff
@@ -162,13 +163,33 @@ public void setBrightness(int ledBrightness) {
             throw new IllegalArgumentException("Brightness needs to be between 0 and "
                     + MAX_BRIGHTNESS);
         }
-        mLedBrightness = ledBrightness;
+        mLedBrightnessGlobal = ledBrightness;
+        mLedBrightness = null;
```

Comment:
why do you make it null? (issues later that you have to null check) Can you just make an empty array / clear it?

Reply:
I make it null to invalidate it, so when the user calls write later, the global value is used instead. This is to allow switching from individual brightness to global brightness.
I am open to do it in any other way, I just thought this one was the simpler one.

Reply:
you could fill the array with the global brightness instead of relying on null to say that there is no individual brightness (Connascence of Values)

Reply:
but the original request for change suggested by jdkoren was to keep them separated.

Reply:
It's not my repo, so it was just a suggestion 👍  I'd go with the mod's advice ;-)